### ISSUE TITLE
Add `keras.ops.copysign`

### ISSUE THREAD

**Issue by Neilblaze** [keras-team/keras#23497]
## Description

`keras.ops` covers the IEEE-754 sign and exponent family fairly completely, such as, `signbit`, `nextafter`, `ldexp`, `hypot`, `heaviside`, `logaddexp`, `logaddexp2` are all there, but `copysign` is missing.

It is available in every framework Keras builds on:

| | available |
|---|---|
| `numpy.copysign` | yes |
| `jax.numpy.copysign` | yes |
| `torch.copysign` | yes |
| `math.copysign` (Python stdlib) | yes |
| `keras.ops.copysign` | **no** |

It is also part of the [Python Array API standard](https://data-apis.org/array-api/latest/API_specification/generated/array_api.copysign.html) (added in the 2023.12 revision), which Keras generally tracks for `keras.ops.numpy`.

### Why it is worth having

`copysign(x1, x2)` returns the magnitude of `x1` with the sign of `x2`. The natural workaround in Keras today is

```python
keras.ops.abs(x1) * keras.ops.sign(x2)
```

which is silently wrong in two ways:

* it returns `0` wherever `x2 == 0`, instead of `±|x1|`;
* it cannot distinguish `-0.0` from `0.0`, so it never produces a negative result for a negative zero.

```python
>>> import numpy as np
>>> np.copysign(3.0, 0.0), np.copysign(3.0, -0.0)
(3.0, -3.0)
>>> abs(3.0) * np.sign(0.0), abs(3.0) * np.sign(-0.0)
(0.0, -0.0)
```

That correctness gap matters wherever sign is carried separately from magnitude including sign-magnitude and symmetric quantization, Householder reflections in QR, round-half-away-from-zero via `trunc(x + copysign(0.5, x))`, and sign-based gradient methods.

### Consistency across backends

Because there is no single op today, users hand-roll it, and the hand-rolled version behaves differently per backend around zero. A single `keras.ops.copysign` gives one definition on TensorFlow, JAX, NumPy, PyTorch and OpenVINO.

NumPy, JAX and PyTorch have a native `copysign` to delegate to. TensorFlow has no `tf.math.copysign`, but it can be composed exactly from the existing `signbit` in the TensorFlow backend. OpenVINO can be composed from `less`/`equal`/`select`/`negative` following the same sign-bit test its existing `signbit` uses.

### Proposal

Add `keras.ops.copysign` / `keras.ops.numpy.copysign`, implemented on all five backends, following `nextafter` for dtype promotion (`result_type(x1, x2, float)`), which matches `jax.numpy.copysign` across the whole dtype matrix.

I have this implemented and tested on all five backends and will open a PR shortly.
